internal/remotestorage/remotestorage.go

```diff
@@ -0,0 +1,65 @@
+package remotestorage
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+// RemoteStorage defines functions to interact with application storage service
+type RemoteStorage interface {
```

Comment:
Separate the interface from its implementation.

Now that you have added the interface, it allows for easier testing from the caller's perspective. However, since the interface and the concrete implementation are both in the same package, the user of the interface is still tied to the implementation via the import of this package.

Reply:
Can I put the implementation in `cli/command/run`? I explored a bit and I think this might be the proper place.

Reply:
Hm. It'll likely be the `vmrunner/cloudrunner/saucerunner` (that doesn't exist yet) that will make use of this functionality and thus require the interface, while `cli/command/run` will be the one to choose and instantiate the concrete implementation.

For the time being, I think you could create a package called `appstore` and put the interface and implementation independent structs in there :)

Reply:
sure. Let me try this way.